test/unittests/commands/xmlupload/test_ontology_client.py

```diff
@@ -0,0 +1,245 @@
+from dataclasses import dataclass
+from pathlib import Path
+from test.unittests.commands.xmlupload.connection_mock import ConnectionMockBase
+from typing import Any
+
+from pytest_unordered import unordered
+
+from dsp_tools.commands.xmlupload.ontology_client import (
+    OntologyClientLive,
+    _get_all_classes_from_graph,
+    _get_all_properties_from_graph,
+    _remove_prefixes,
+    deserialize_ontology,
+)
+
+# pylint: disable=missing-class-docstring,missing-function-docstring,unused-argument,redefined-outer-name,protected-access
+
+
+@dataclass
+class ConnectionMock(ConnectionMockBase):
+    get_responses: dict[Any, Any]
+
+    def get(self, route: str, headers: dict[str, str] | None = None) -> dict[Any, Any]:
+        return self.get_responses
+
+
+class TestGetAllClassesFromGraph:
+    @staticmethod
+    def test_single_class() -> None:
```

Comment:
For my taste, this is too fine-grained. As I commented in `_get_all_classes_from_graph`, I would merge the 2 functions, and then test the higher-level one.